lib/vmdb/initializer.rb

```diff
@@ -17,22 +17,35 @@ def self.init
     def self.init_secret_token
       return if Rails.application.config.secret_key_base
 
-      token = if ActiveRecord::Base.connectable? && MiqDatabase.table_exists? && MiqDatabase.any?
-        MiqDatabase.first.session_secret_token
-      else
-        SecureRandom.hex(64)
-      end
+      token = session_secret_token || SecureRandom.hex(64)
 
       Rails.application.config.secret_key_base = token
     end
 
+    private_class_method def self.session_secret_token
+      return nil unless ActiveRecord::Base.connectable?
+      return nil unless MiqDatabase.table_exists?
+      return nil unless MiqDatabase.any?
+
+      begin
+        MiqDatabase.first.session_secret_token
+      rescue ManageIQ::Password::PasswordError => err
```

Comment:
Should I add a test that raises this and returns a random hex from init_secret_token?  Feels heavily mocked but can if it makes sense.  I'm torn.

Reply:
You can just put a bad value in the column directly, and then let it blow up.

Reply:
Something like

```
> MiqDatabase.first.tap { |d| d.write_attribute(:session_secret_token, "xxx"); d.save!(validate: false) }
> MiqDatabase.first.session_secret_token
/Users/jfrey/.gem/ruby/3.3.7/gems/manageiq-password-1.2.0/lib/manageiq/password.rb:48:in `decrypt': cannot decrypt plaintext string (ManageIQ::Password::PasswordError)
```

Reply:
haha, awesome, good idea

Reply:
Ok, done.  Added a test and cleaned up the output.